hibernate-core/src/main/java/org/hibernate/persister/collection/BasicCollectionPersister.java

```diff
@@ -501,16 +525,28 @@ private void applyUpdateRowValues(
 				0,
 				jdbcValueBindings,
 				null,
-				(valueIndex, bindings, y, jdbcValue, jdbcValueMapping) -> {
-					if ( !jdbcValueMapping.isUpdateable() || jdbcValueMapping.isFormula() ) {
-						return;
-					}
-					bindings.bindValue(
-							jdbcValue,
-							jdbcValueMapping,
-							ParameterUsage.SET
-					);
-				},
+				hasIndex() ?
+						(valueIndex, bindings, y, jdbcValue, jdbcValueMapping) -> {
+							if ( jdbcValueMapping.isFormula() ) {
+								return;
+							}
+							bindings.bindValue(
+									jdbcValue,
+									jdbcValueMapping,
+									ParameterUsage.SET
+							);
+						}
+						:
+						(valueIndex, bindings, y, jdbcValue, jdbcValueMapping) -> {
+							if ( !jdbcValueMapping.isUpdateable() || jdbcValueMapping.isFormula() ) {
```

Comment:
Shouldn't this be enough since formulas are never updatable?

```suggestion
							if ( !jdbcValueMapping.isUpdateable() ) {
```

Reply:
it should but not 100% sure it is. Checking....

Reply:
it seems that at the moment `isFormula` does not always imply `!isUpdatable`.

Reply:
It should maybe, wdyt?

Reply:
yes but it seems to me a topic unrelated to this issue, probably better to create a separate Jira